### ISSUE TITLE
Reasoning for choosing VoxelMap instead of other mod

### ISSUE THREAD

**Issue by jamezrin** [xJon/The-1.12.2-Pack#488]
It's not the first time I see someone having issues with VoxelMap crashing and it has happened to me. It is quite honestly, a pain in the ass. 

Out of nowhere it crashes. At least there is something in this modpack that makes it so that crashes don't outright close your game, but still.

Every time someone has this problem with VoxelMap, the suggestion to fix it is to "remove" it, and quite honestly it's the best workaround, because VoxelMap crashes no matter what, at least that is my experience with VoxelMap and this modpack.

The point of this issue is this question: 

Why isn't VoxelMap swapped with something more stable already? Like Xaero's Minimap or JourneyMap. 
I have started using Xaero's Minimap recently and I think it is a very capable replacement for VoxelMap. 
Unfortunately the waypoints aren't imported automatically, I had to do the conversion manually by myself, I haven't had any crashes since I removed VoxelMap.

I wrote a little script to convert VoxelMap waypoints to Xaero's Minimap ones, if someone finds it useful there you go https://github.com/jamezrin/voxel2xaero/

**Comment by xJon** [comment]
As you mentioned yourself, one of the problems with switching the minimap mod of a modpack is players losing their map data & waypoints. I'm happy to see you offer such script although I doubt all players will know how to run Python 3 scripts.
I think that if there was an executable batch file it would make it easier for most players.

That being said, I do agree with you. VoxelMap for 1.12.2 has been very problematic, it's been crashing randomly to many players and it's been incompatible with Botania. I personally tried *many* times to reach out to VoxelMap's author about the issues, and got minimal respond without any clear intent to resolve things.

I've been hopeful for months now that the author will get the issues fixed but as they seem to more or less ignore my requests, I've been thinking more about a proper replacement.
Additionally, the latest update of VoxelMap includes a *cache location change*, without any automatic transfer mechanism, which means players who don't necessarily know how to transfer it to the right place would lose all of their data; just like in a minimap mod change.

All of the above means that most likely we will see a minimap mod change in version `1.3.0`, but not without a community poll.
I know that a lot of players like VoxelMap and its been The X Packs' signature minimap mod for years, so I'd like to see a majority agreeing for a change before it'd take place.

**Comment by xJon** [comment]
In fact, I will write a batch script to interact with your python script so players using Windows 10 could use it more easily.

If we do end up switching to Xaero's, would you mind if I include your script in the Technic distribution of the modpack?

**Comment by jamezrin** [comment]
> That being said, I do agree with you. VoxelMap for 1.12.2 has been very problematic, it's been crashing randomly to many players and it's been incompatible with Botania. I personally tried _many_ times to reach out to VoxelMap's author about the issues, and got minimal respond without any clear intent to resolve things.

Yeah, it's a shame this is happening... I really liked VoxelMap, it has a feature I really miss in Xaero (highlight waypoints), but crashes are just unbearable.

> All of the above means that most likely we will see a minimap mod change in version `1.3.0`, but not without a community poll.
> I know that a lot of players like VoxelMap and its been The X Packs' signature minimap mod for years, so I'd like to see a majority agreeing for a change before it'd take place.

Good idea :+1: 

> In fact, I will write a batch script to interact with your python script so players using Windows 10 could use it more easily.
> 
> If we do end up switching to Xaero's, would you mind if I include your script in the Technic distribution of the modpack?

You are more than welcome to do so, in any way you want. Though I have to say I haven't tried it with Windows 10, but it should work fine.

**Comment by xJon** [comment]
@jamezrin Apart from the waypoints, do you know anything about the possibility of world map data transfer between the two mods?

**Comment by jamezrin** [comment]
Are you referring to Xaero's World Map? I didn't know it existed, to be honest I didn't really use the world map in VoxelMap that often so I just took it as a compromise for not crashing.
I assume it is possible but it would be much more involved, I haven't tried Xaero's World Map yet, so I don't know its structure.

**Comment by xJon** [comment]
Yes, I was referring to Xaero's World Map. No worries, thanks anyway

**Comment by jamezrin** [comment]
I have started using Xaero's World Map, and unless I missed something, the only data that Xaero's World Map stores is cache of the terrain, and I'm not sure if something to convert the cache in VoxelMap to Xaero's World Map would be useful. 
The structures are bastly different, making it very if not impossible to do externally. 

I opened my map and it had a bunch of loading marks, and after a few seconds the terrain started to render.

**Comment by xJon** [comment]
Do you have better insight on reproducing the issues with VoxelMap?
I'd like to test them on its [new update](https://www.curseforge.com/minecraft/mc-mods/voxelmap/files/3017424).

**Comment by jamezrin** [comment]
I tried it for a few hours and it hasn't crashed so far...
It was kinda random for me honestly, sometimes it would not happen for a few hours and sometimes it would crash 2 or 3 times in the span of 15 minutes.
I don't have any mana pools in my world so that's something to consider, my crashes were related to light level calculation or whatever

**Comment by xJon** [comment]
It'd be ideal to update VoxelMap instead of switching it out. I will continue to test the new build

**Comment by xJon** [comment]
I was still unable to reproduce crashes while using the latest update so this will stay on hold for now.

**Comment by xJon** [comment]
The issue still seem to be preset yet it is helped by `1.2.1` changes with the Forge config to automatically remove erroring tile entities.
Unfortunately this only helps the Botania incompatibility and not the other crashes.

**Comment by xJon** [comment]
All mentioned issues should be resolved in `1.5.1` thanks to https://github.com/ACGaming/VoxelMapFixes :)
